File: Selenium/Activities/Actiivty16.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import Select

# Set up the driver
driver = webdriver.Chrome()

# Open the selects page
driver.get("https://training-support.net/webelements/selects")

# Print the title of the page
print("Page title:", driver.title)

# Find the Single Select dropdown
dropdown = driver.find_element(By.ID, "singleSelect")
select = Select(dropdown)

# Select the second option using visible text
select.select_by_visible_text("Option 2")

# Select the third option using index (index starts at 0)
select.select_by_index(2)

# Select the fourth option using value
select.select_by_value("4")

# Get all options and print them
options = select.options
print("Dropdown options:")
for option in options:
    print(option.text)

# Close the browser
driver.quit()

File: Selenium/Activities/Actiivty16.java
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.support.ui.Select;
import java.util.List;

public class Activity16 {
    public static void main(String[] args) {
        // Set up the driver
        WebDriver driver = new ChromeDriver();

        // Open the selects page
        driver.get("https://training-support.net/webelements/selects");

        // Print the title of the page
        System.out.println("Page title: " + driver.getTitle());

        // Find the Single Select dropdown
        WebElement dropdown = driver.findElement(By.id("singleSelect"));
        Select select = new Select(dropdown);

        // Select the second option using visible text
        select.selectByVisibleText("Option 2");

        // Select the third option using index (index starts at 0)
        select.selectByIndex(2);

        // Select the fourth option using value
        select.selectByValue("4");

        // Get all options and print them
        List<WebElement> options = select.getOptions();
        System.out.println("Dropdown options:");
        for (WebElement option : options) {
            System.out.println(option.getText());
        }

        // Close the browser
        driver.quit();
    }
}
